### ISSUE TITLE
Model Configuration Guide Updates and Loss Layers Usage Examples Request

### ISSUE THREAD

**Issue by kim-com** [nnstreamer/nntrainer#2826]
1. Update request for model configuration guide
When looking at the items in the ini file generated after saving the model, there are many things that are not mentioned in the [official documentation](https://github.com/nnstreamer/nntrainer/blob/main/docs/configuration-ini.md).
For example, among the following contents, only `type`, `epochs`, `loss`, `save_path`, and `batch_size` have descriptions, but the rest do not.

```ini
[model]
batch_size = 32
continue_train = false
epochs = 200
loss = mse
loss_scale = 1.000000
memory_optimization = true
memory_swap = false
memory_swap_lookahead = 0
memory_swap_path = .
model_tensor_type = FP32-FP32
tensor_format = NCHW
type = NeuralNetwork
```

2. Guide and examples for using loss layers
When loading a model, I received a log message like this: 
```text
[NNTRAINER WARN  2024-12-11 14:29:44] (/home/-----/nntrainer/nntrainer/models/model_common_properties.cpp:isValid:22) Model loss property is deprecated, use loss layer directly instead
```
It would be helpful to have some examples on how to use loss layers.

3. When saving a model as an ini file, the loss_scale value was 0, and the following error occurred when loading it:

code:
```c++
try {
    model->load(config, ml::train::ModelFormat::MODEL_FORMAT_INI);
  } catch (std::exception &e) {
    std::cerr << "Error while loading model! details: " << e.what()
              << std::endl;
    return 1;
  }
```
bash:
```bash
Error while loading model! details: invalid argument from c style throw
```
log:
```text
[NNTRAINER ERROR 2024-12-11 15:50:30] (/home/kimhyeongjun/tizen/github/workspace2/nntrainer/nntrainer/models/model_common_properties.cpp:isValid:48) Loss scale cannot be 0
[NNTRAINER ERROR 2024-12-11 15:50:30] (/home/kimhyeongjun/tizen/github/workspace2/nntrainer/nntrainer/models/model_loader.cpp:loadModelConfigIni:131) St16invalid_argument argument is not valid
```

**Comment by EunjuYang** [comment]
Hello, @kim-com 
Thank you for your concern about NNTrainer project 😊
NNTrainer has been actively updated, and the very recent features (e.g., memory swap and loss scaling) are not yet updated in the documentation. We will update them soon.
About the second issue, we're going to investigate it. Thank you.
